sys/compat/linux/linux_misc.c

```diff
@@ -2963,3 +2966,124 @@ linux_ioprio_set(struct thread *td, struct linux_ioprio_set_args *args)
 	}
 	return (error);
 }
+
+/* The only flag is O_NONBLOCK */
+#define B2L_MQ_FLAGS(bflags)	((bflags) != 0 ? LINUX_O_NONBLOCK : 0)
+#define L2B_MQ_FLAGS(lflags)	((lflags) != 0 ? O_NONBLOCK : 0)
+
+int
+linux_mq_open(struct thread *td, struct linux_mq_open_args *args)
+{
+	struct mq_attr attr;
+	int error, flags;
+
+	flags = linux_common_openflags(args->oflag);
+	if ((flags & O_ACCMODE) == O_ACCMODE || (flags & O_EXEC) != 0)
+		return (EINVAL);
+	flags = FFLAGS(flags);
+	if ((flags & O_CREAT) != 0 && args->attr != NULL) {
+		error = copyin(args->attr, &attr, sizeof(attr));
+		if (error != 0)
+			return (error);
+		attr.mq_flags = L2B_MQ_FLAGS(attr.mq_flags);
+	}
+
+	return (kern_kmq_open(td, args->name, flags, args->mode,
+	    args->attr != NULL ? &attr : NULL));
+}
+
+int
+linux_mq_unlink(struct thread *td, struct linux_mq_unlink_args *args)
+{
+	struct kmq_unlink_args bsd_args = {
+		.path = PTRIN(args->name)
+	};
+
+	return (sys_kmq_unlink(td, &bsd_args));
```

Comment:
Still you are using sys_kmq... instead of adding yet another kern_kmq function

Reply:
> Still you are using sys_kmq... instead of adding yet another kern_kmq function

In this case we can because a pointer is the only argument mq_unlink() uses, and sys_kmq_unlink does the heavy-lifting well...